src/west/app/project.py

```diff
@@ -865,10 +865,15 @@ def _update_concurrent_thread(args):
             saved_stdout = os.dup(sys.stdout.fileno())
             saved_stderr = os.dup(sys.stderr.fileno())
 
+            logfile = os.path.join(logdir, project.name)
+
+            # the name might contain `/`
+            os.makedirs(os.path.dirname(logfile), exist_ok=True)
+
             # the dup2s replace stderr and stdout to a file.
             # unlike just replacing the stream objects this also affects writes
             # by child processes.
-            logfile = open(os.path.join(logdir, project.name), "wb")
+            logfile = open(logfile, "wb")
```

Comment:
No big deal but I'm not a fan of the same variable changing type like this, I think it's a bit confusing. `logfilepath` instead? `path` also highlights the fact that it's not just a basename.